src/amo/reducers/viewContext.js

```diff
@@ -0,0 +1,33 @@
+/* @flow */
+import {
+  ADDON_TYPE_EXTENSION,
+  ADDON_TYPE_THEME,
+  VIEW_CONTEXT_EXPLORE,
+  VIEW_CONTEXT_HOMEPAGE,
+  VIEW_CONTEXT_SET,
+} from 'core/constants';
+
+
+export type ViewContextType = {|
+  context: ADDON_TYPE_EXTENSION | ADDON_TYPE_THEME | VIEW_CONTEXT_EXPLORE |
+    VIEW_CONTEXT_HOMEPAGE,
+|}
+
+export type ViewContextActionType = {|
+  type: string,
+  payload: ViewContextType,
+|}
+
+export const initialState = { context: VIEW_CONTEXT_EXPLORE };
+
+export default function viewContext(
+  state : ViewContextType = initialState, action : ViewContextActionType
+) {
+  const { payload } = action;
+  switch (action.type) {
+    case VIEW_CONTEXT_SET:
```

Comment:
So the actual type I feel is handy to name with the reducer first solely for the purpose of organisation in the constants file. You have all the `CATEGORY_$VERB` actions grouped together naturally that way. I think we should change them all to make action verb first (so `SET_VIEW_CONTEXT` over `VIEW_CONTEXT_SET` and `FETCH_CATEGORIES` over `CATEGORIES_FETCH`) but I think we should do it en-masse and organise the constants so each reducer's actions are someplace.

That said the confusion for `VIEW_CONTEXT_SET` being similar to `VIEW_CONTEXT_HOMEPAGE` is something I noticed myself and since you brought it up too I'll change it in this patch 😄